HLTrigger/Egamma/plugins/HLTPMMassFilter.cc

```diff
@@ -95,25 +97,18 @@ bool HLTPMMassFilter::hltFilter(edm::Event& iEvent,
       charge.push_back(refele->charge());
     }
 
+    std::vector<bool> save_cand(electrons.size(), true);
     for (unsigned int jj = 0; jj < electrons.size(); jj++) {
-      TLorentzVector p1Ele = pEleCh1.at(jj);
       for (unsigned int ii = jj + 1; ii < electrons.size(); ii++) {
-        TLorentzVector p2Ele = pEleCh1.at(ii);
-
-        if (fabs(p1Ele.E() - p2Ele.E()) < 0.00001)
-          continue;
         if (reqOppCharge_ && charge[jj] * charge[ii] > 0)
           continue;
-
-        TLorentzVector pTot = p1Ele + p2Ele;
-        double mass = pTot.M();
-
-        if (mass >= lowerMassCut_ && mass <= upperMassCut_) {
+        if (isGoodPair(pEleCh1[jj], pEleCh1[ii])) {
           n++;
-          refele = electrons[ii];
-          filterproduct.addObject(TriggerElectron, refele);
-          refele = electrons[jj];
-          filterproduct.addObject(TriggerElectron, refele);
+          for (auto const idx : {jj, ii}) {
+            if (save_cand[idx])
+              filterproduct.addObject(TriggerElectron, electrons[idx]);
+            save_cand[idx] = false;
+          }
```

Comment:
In the `isElectron1_ && isElectron2_` case, there is no change in how candidate pairs are counted (as the nested loop was already considering only unique pairs). The change is that the same `TriggerElectron` is not added twice the output product.